draco/analysis/powerspectrum.py

```diff
@@ -69,13 +71,110 @@ def process(self, klmodes):
 
         q_list = []
 
+        st = time.time()
+
         for mi, m in klmodes.vis[:].enumerate(axis=0):
             ps_single = pse.q_estimator(m, klmodes.vis[m, : klmodes.nmode[m]])
             q_list.append(ps_single)
 
         q = klmodes.comm.allgather(np.array(q_list).sum(axis=0))
         q = np.array(q).sum(axis=0)
 
+        et = time.time()
+        if mpiutil.rank0:
+            print("m, time needed for quadratic pse", m, et - st)
+
+        # reading from directory
+        fisher, bias = pse.fisher_bias()
+
+        ps = containers.Powerspectrum2D(
+            kperp_edges=pse.kperp_bands, kpar_edges=pse.kpar_bands
+        )
+
+        npar = len(ps.index_map["kpar"])
+        nperp = len(ps.index_map["kperp"])
+
+        # Calculate the right unmixing matrix for each ps type
+        if self.pstype == "unwindowed":
+            M = la.pinv(fisher, rcond=1e-8)
+        elif self.pstype == "uncorrelated":
+            Fh = la.cholesky(fisher)
+            M = la.inv(Fh) / Fh.sum(axis=1)[:, np.newaxis]
+        elif self.pstype == "minimum_variance":
+            M = np.diag(fisher.sum(axis=1) ** -1)
+
+        ps.powerspectrum[:] = np.dot(M, q - bias).reshape(nperp, npar)
+        ps.C_inv[:] = fisher.reshape(nperp, npar, nperp, npar)
+
+        return ps
+
+
+class QuadraticPSEstimationXLarge(QuadraticPSEstimation):
```

Comment:
Why does this need to be a separate class at all? Is there any reason this can't just replace the existing implementation? Or if not, merge them and change the behaviour to a config option, or even better, something that gets automatically determined based on the input data size.

Reply:
I mean, I could put them into one class and add a config option, but they two classes are fundamentally very different for the following reasons:
- the `QuadraticPSEstimationLarge` uses from `driftscan` the `PSMontelCarloLarge` class in which the clarray and the qestimator are MPIarrays parallelized over power spectrum bands. This is very different from the original code, in which clarray is gathered on one rank and qestimators are calculated locally. 
- This class respects the fact that clarray and qestimator are parallelized over nbands, therefore the sky data need to be pre-computed for all m first as each rank needs all m's. This is also very different from the original code, where the sky data were handled parallel in m. 

If we were to use a config option - then we would have one class but and if/else statement in the process method handling the two different calculations. 

I personally thought a separate class is cleaner, but after all - you are the boss ;)